### PR TITLE
fix: add multiple blackbox tests and fix small bugs

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
As we start to integrate more and more inputs, it is important that we test actual real documents from users, and best way I can see is to add them as BlackBox tests. The BlackBox tests are focused on catching syntax errors and not logical errors.

It adds the following documents:
- AsyncAPI 2.0 - https://github.com/jonaslagoni/generator-model-sdk/tree/feature/new_blackbox_tests/test/blackbox/docs/AsyncAPI-2_0
- AsyncAPI 2.1 - https://github.com/jonaslagoni/generator-model-sdk/tree/feature/new_blackbox_tests/test/blackbox/docs/AsyncAPI-2_1
- JSON Schema draft 7 - https://github.com/jonaslagoni/generator-model-sdk/tree/feature/new_blackbox_tests/test/blackbox/docs/JsonSchemaDraft-7

This PR fixes some small issues that was discovered:
- Encountering union types, can be rendered in some cases, at least until proper union types can be rendered, so we do "best effort".
- CommonModel types for the languages, should never use the input as-is.
- Java renderer did not include dependency when `patternProperties` were used without `additionalProperties`. 
- TS apparently have `arguments` as a reserved keyword.

### CODE REVIEW

File: src/generators/java/JavaGenerator.ts [hunk 1]
```diff
@@ -29,10 +29,13 @@ export class JavaGenerator extends AbstractGenerator<JavaOptions> {
 
   render(model: CommonModel, inputModel: CommonInputModel): Promise<RenderOutput> {
     const kind = TypeHelpers.extractKind(model);
-    if (kind === ModelKind.ENUM) {
+    //We dont support union in Go generator, however, if union is an object, we render it as a struct.
```
Comment (magicmatatjahu) [inline src/generators/java/JavaGenerator.ts:32]:
```suggestion
    // We don't support union in Java generator, however, if union is an object, we render it as a class.
```

File: src/generators/go/GoGenerator.ts [hunk 2]
```diff
@@ -55,12 +55,15 @@ export class GoGenerator extends AbstractGenerator<GoOptions> {
   render(model: CommonModel, inputModel: CommonInputModel): Promise<RenderOutput> {
     const kind = TypeHelpers.extractKind(model);
     switch (kind) {
+    case ModelKind.UNION:
+      //We dont support union in Go generator, however, if union is an object, we render it as a struct.
```
Comment (magicmatatjahu) [inline src/generators/go/GoGenerator.ts:59]:
```suggestion
      // We don't support union in Go generator, however, if union is an object, we render it as a struct.
```

File: test/blackbox/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Running the tests
+The tests can either be run by installing all dependencies locally, or running it through docker.
+
+To run the BlackBox tests through Docker, run the command `npm run docker:test:blackbox`.
+
+If you want to run the BlackBox tests locally, you have to install a couple of dependencies:
+- To to run the `java` BlackBox tests, you need to have JDK installed.
+- To to run the `ts` BlackBox tests, you need to have TypeScript installed globally - `npm install -g typescript`.
+- To to run the `C#` BlackBox tests, you need to have C# compiler installed globally. - https://www.mono-project.com/download/stable/
```
Comment (magicmatatjahu) [inline test/blackbox/README.md:20]: Please also add info about `Go` language :)

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: test/blackbox/README.md (hunk 3 above)
Comment (jonaslagoni) [reply test/blackbox/README.md:20]: 🤦 right!

**Review by magicmatatjahu — APPROVED** [review]
For me, it looks good. @alequetzalli Do you want to make review of changed mds?

File: docs/development.md [hunk 4]
```diff
@@ -7,24 +7,15 @@ You can either build the image and run the needed commands manually or rather us
...
-If you prefer to install all the dependencies locally, keep reading.
-
-The blackbox testing have some different requirements in order to successfully run:
-- To to run the `java` blackbox tests, you need to have JDK installed.
-- To to run the `ts` blackbox tests, you need to have TypeScript installed globally - `npm install -g typescript`.
-- To to run the `C#` blackbox tests, you need to have C# compiler installed globally. - https://www.mono-project.com/download/stable/
-
-By default the blackbox tests are not run with the regular `npm run test`, but can be run with `npm run test:blackbox`.
+## BlackBox testing
+We have a bunch of BlackBox tests, that are run separately to the `npm run test` script. Please refer to the [BlackBox documentation](../test/blackbox) for further information.
```
Comment (alequetzalli) [inline docs/development.md:21]:
```suggestion
We have several BlackBox tests that are run separately from the `npm run test` script. Please refer to the [BlackBox documentation](../test/blackbox) for further information.
```

File: test/blackbox/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# BlackBox tests
+
+As we start to integrate more and more inputs, it is important that we test actual real inputs from users to ensure that the library behaves as expected. The best way I can see is to add them as BlackBox tests. The BlackBox tests are focused on catching syntax errors and not logical errors, although they can be manually used used for this.
```
Comment (alequetzalli) [inline test/blackbox/README.md:3]:
```suggestion
As we start to integrate more and more inputs, it is important that we test actual and real inputs from users to ensure the library behaves as expected. For now, the best way is to add them as BlackBox tests. The BlackBox tests are focused on catching syntax errors and not logical errors, although they can be manually used for this.
```

File: test/blackbox/README.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+To run the BlackBox tests through Docker, run the command `npm run docker:test:blackbox`.
+
+If you want to run the BlackBox tests locally, you have to install a couple of dependencies:
+- To to run the `Java` BlackBox tests, you need to have JDK installed.
+- To to run the `TypeScript` BlackBox tests, you need to have TypeScript installed globally - `npm install -g typescript`.
+- To to run the `C#` BlackBox tests, you need to have C# compiler installed globally. - https://www.mono-project.com/download/stable/
+- To to run the `Go` BlackBox tests, you need to have GoLang installed - https://golang.org/doc/install
+
+By default the BlackBox tests are not run with the regular `npm run test`, but can be run with `npm run test:blackbox`.
```
Comment (alequetzalli) [inline test/blackbox/README.md:23]:
```suggestion
By default, the BlackBox tests are not run with the regular `npm run test`, but can be run with `npm run test:blackbox`.
```

**Comment by alequetzalli** [comment]
> For me, it looks good. @alequetzalli Do you want to make review of changed mds?

dun dun dun... Poor @jonaslagoni 😂.. yes! I accept and gave a succint review! 😆😀🦄✨✨

**Review by alequetzalli — APPROVED** [review]
LGTM 🚀 ✨✨✨

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**